### DISCUSSION TITLE
PG::UndefinedFunction: ERROR:  operator does not exist: bigint = uuid

### DISCUSSION THREAD

**Discussion by awesome** [jonhue/acts_as_favoritor#226] [Show and tell]
**Describe the bug**

```
PG::UndefinedFunction: ERROR:  operator does not exist: bigint = uuid
LINE 1: ...table_type" = $1 AND "favorites"."favoritable_id" = "partner...

```

**To Reproduce**

Use Postgres `uuid` extension.
```
# db/migrate/20210313001820_enable_postgres_extensions.rb
class EnablePostgresExtensions < ActiveRecord::Migration[6.1]
  def change
    enable_extension 'pgcrypto'
  end
end
```

Default acts_as_favoritor generator schema output:
```
# db/schema.rb
create_table "favorites", force: :cascade do |t|
  t.string "favoritable_type", null: false
  t.bigint  "favoritable_id", null: false
  t.string "favoritor_type", null: false
  t.bigint  "favoritor_id", null: false
  t.string "scope", default: "favorite", null: false
[…]
end

```

**Fix**

After generating migration, before `rails db:migrate`:
1. Add `id: :uuid` to block arguments.
2. Append `t.references…` lines with `type: :uuid` 

```
create_table :favorites, id: :uuid, force: true do |t|
  t.references :favoritable, polymorphic: true, null: false, type: :uuid
  t.references :favoritor, polymorphic: true, null: false, type: :uuid
```

**Expected behavior**

```
# db/schema.rb
create_table :favorites, id: :uuid, force: true do |t|
  t.string "favoritable_type", null: false
  t.uuid "favoritable_id", null: false
  t.string "favoritor_type", null: false
  t.uuid "favoritor_id", null: false
  t.string "scope", default: "favorite", null: false
[…]
end

```

**Environment**

Rails 6.1

**Comment by jonhue** [comment]
Is this a bug? When using the uuid plugin, you can just change the generator output.

**Comment by awesome** [comment]
Not really a bug, but a solution for using acts_as_favoritor with Postgres UUID extension. Didn't work OTB, so found a solution and recording it here. Zero search results for uuid in the repo: https://github.com/jonhue/acts_as_favoritor/search?q=uuid

Thanks for the tip "you can just change the generator output." Sounds great, but still don't know how to do that.

**Reply by jonhue** [reply]
yeah, I just meant that you can edit the files after they were generated. The generator is just supposed to be a utility to make the setup process a little easier, anyway.

**Reply by awesome** [reply]
Thanks for the clarification. Useful gem, great work! I'm going to prod with it.
